parsec/backend/postgresql/handler.py

```diff
@@ -166,15 +176,32 @@ async def init(self, nursery):
         self._task_status = await start_task(nursery, self._run_connections)
 
     async def _run_connections(self, task_status=trio.TASK_STATUS_IGNORED):
-        async with triopg.create_pool(
-            self.url, min_size=self.min_connections, max_size=self.max_connections
-        ) as self.pool:
-            # This connection is dedicated to the notifications listening, so it
-            # would only complicate stuff to include it into the connection pool
-            async with triopg.connect(self.url) as self.notification_conn:
-                await self.notification_conn.add_listener("app_notification", self._on_notification)
-                task_status.started()
-                await trio.sleep_forever()
+        while True:
+            try:
+                async with triopg.create_pool(
+                    self.url, min_size=self.min_connections, max_size=self.max_connections
+                ) as self.pool:
+                    # This connection is dedicated to the notifications listening, so it
+                    # would only complicate stuff to include it into the connection pool
+                    async with triopg.connect(self.url) as self.notification_conn:
+                        await self.notification_conn.add_listener(
+                            "app_notification", self._on_notification
+                        )
+                        task_status.started()
+                        await trio.sleep_forever()
+            except OSError as err:
+                if self.first_tries_number == 1:
+                    raise err
```

Comment:
can use `raise` without err (this way we can also have `except OSError:`)

Reply:
Changed